Add documentation about RocksDB usage (#827)
rsksmart/devportal: _rsk/node/configure/index.md

[GET STARTED](http://rocksdb.org/docs/getting-started.html).

Adventages (taken from source):
* RocksDB uses a log structured database engine, written entirely in C++, for maximum performance. Keys and values are just arbitrarily-sized byte streams.
* RocksDB is optimized for fast, low latency storage such as flash drives and high-speed disk drives. RocksDB exploits the full potential of high read/write rates offered by flash or RAM.
* RocksDB is adaptable to different workloads. From database storage engines such as [MyRocks](https://github.com/facebook/mysql-5.6) to [application data caching](http://techblog.netflix.com/2016/05/application-data-caching-using-ssds.html) to embedded workloads, RocksDB can be used for a variety of data needs.
* RocksDB provides basic operations such as opening and closing a database, reading and writing to more advanced operations such as merging and compaction filters.

In order to take advange if it's high performance, we added this experimental feature to our app. It could be used in the `*.conf` file by just setting `keyvalue.datasource=rocksdb`.

`keyvalue.datasource` supports either: `rocksdb` or `leveldb`

> It is good idea to reset the database everytime you want to switch between different db storage, for instance from `leveldb` to `rocksdb` or viceversa, so that way the data is stored in the format of the selected DB.
> 
> Sample command using `rocksdb` when the previous db was `leveldb` (notice we add `--import` at the end to reset and re-import the database):

Reviewer: ```suggestion
If you wish to switch between the different storage options,
for example from `leveldb` to `rocksdb` or vice versa, 
you must **restart** the node with the import option.

The following sample command show how to do this when
the RSKj node was previously running the default (`leveldb`),
and wants to run with `rocksdb` next.
Note the use of the `--import` flag, which resets and re-imports the database.
```

- Running the node with the `java` command, add `-Drsk.conf.file=path/to/your/file.conf`
- Compiling the node with IntelliJ, add to VM options: `-Drsk.conf.file=path/to/your/file.conf`

> ### Using RocksDb (Experimental)

Reviewer 2: ```suggestion
### Using RocksDB (Experimental)
```

By default, RSKj runs using [LevelDB](https://dbdb.io/db/leveldb).
There is an option to use an alternate storage option,
[RocksDB](http://rocksdb.org/), instead.

> The RocksDb option was introduced to enable higher performance

Reviewer 2: ```suggestion
The RocksDB option was introduced to enable higher performance
```

> The RocksDb option was introduced to enable higher performance

Reviewer 2: ```suggestion
The RocksDB option was introduced to enable higher performance
```

within the RSKj nodes.

[RocksDB](http://rocksdb.org/) is a persistent key-value store for fast storage environments
RocksDB is an embeddable persistent key-value store for fast storage.

[GET STARTED](http://rocksdb.org/docs/getting-started.html).

> Adventages (taken from source):

Reviewer 2: ```suggestion
Advantages (taken from source):
```

* RocksDB uses a log structured database engine, written entirely in C++, for maximum performance. Keys and values are just arbitrarily-sized byte streams.
* RocksDB is optimized for fast, low latency storage such as flash drives and high-speed disk drives. RocksDB exploits the full potential of high read/write rates offered by flash or RAM.
* RocksDB is adaptable to different workloads. From database storage engines such as [MyRocks](https://github.com/facebook/mysql-5.6) to [application data caching](http://techblog.netflix.com/2016/05/application-data-caching-using-ssds.html) to embedded workloads, RocksDB can be used for a variety of data needs.
* RocksDB provides basic operations such as opening and closing a database, reading and writing to more advanced operations such as merging and compaction filters.

> ### How to use RocksDb

Reviewer 2: ```suggestion
### How to use RocksDB
```

Modify the relevant RSKj config file (`*.conf`) file
and set the property `keyvalue.datasource=rocksdb`.

The `keyvalue.datasource` property in the config
may only be either `rocksdb` or `leveldb`.

If you wish to switch between the different storage options,
for example from `leveldb` to `rocksdb` or vice versa, 
you must **restart** the node with the import option.

> The following sample command show how to do this when

Reviewer 2: ```suggestion
The following sample command shows how to do this when
```
